src/users/src/users-service/routes.go

```diff
@@ -47,6 +49,30 @@ var routes = Routes{
 		"/users/identityid/{identityID}",
 		UserShowByIdentityId,
 	},
+	Route{
+		"GetFilteredUser",
+		"GET",
+		"/users/filter/",
+		GetFilteredUser,
+	},
+	Route{
+		"GetRandomUser",
+		"GET",
+		"/users/getRandomUser/",
+		GetRandomUser,
+	},
+	Route{
+		"ClaimUser",
+		"POST",
+		"/users/claim/",
```

Comment:
I'd prefer `PUT /users/{userID}/claim` to keep it tied to a specific user resource. The identityId can be passed as a JSON payload.

Reply:
Actually, we dont really need the identityId to be passed at all since we just want to claim a shopper and there is no need to support the entire audit of which identity is used by which shopper. Updated the method